hearth/lib/hearth/http/header_list_builder.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module Hearth
+  module Http
+    # @api private
+    module HeaderListBuilder
+      class << self
+        # builds a string from a list of possibly quoted values
+        # ensures that quoted values are escaped
+        # could name the below to be join_string_list?
+        def build_string_list(value)
+          unless value.respond_to?(:compact)
+            raise ArgumentError,
```

Comment:
I think you can remove validation here - it's a private class and will be controlled by codegen. This case will never arise.